### PR TITLE
Switch to serde_derive

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/430)

### CODE REVIEW

**Comment by nox** [comment]
@alexcrichton There is an issue with default features again, somehow running `cargo build --no-default-features --features serde_derive` enables the `with-syntex` feature of `serde_codegen`. Any idea whether this is fixable?

**Comment by alexcrichton** [comment]
@nox features are unioned amongst crates, so it's likely that some other crate is enabling the `with-syntex` feature of `serde_codegen`. As to debugging *which* crate is enabling that, unfortunately we don't have a lot of tooling for that right now.

**Comment by nox** [comment]
> so it's likely that some other crate is enabling the with-syntex feature of serde_codegen

AFAIK no, if I comment the `default` and `codegen` features of `webrender_traits`, `with-syntex` isn't enabled anymore. Will look again though.

**Comment by nox** [comment]
@alexcrichton Did you check the Cargo.lock file? I just noticed that while syntex_syntax is present in it, I see no "Compiling syntex_syntax" message.

File: webrender/Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 [package]
 name = "webrender"
-version = "0.5.1"
+version = "0.6.0"
```
Comment (Manishearth) [inline webrender/Cargo.toml]: I don't see why this version bump is necessary.

**Comment by nox** [comment]
@bors-servo r=Manishearth

As discussed on IRC.

**Pull request merged**